src/services/search/expressions/note_content_fulltext.ts

```diff
@@ -133,6 +133,26 @@ class NoteContentFulltextExp extends Expression {
 
             content = content.replace(/&nbsp;/g, ' ');
         }
+        else if (type === 'mindMap' && mime === 'application/json') {
+            let mindMapcontent = JSON.parse (content);
+            let topic = mindMapcontent.nodedata.topic;
+            content = utils.normalize(topic.toString());
```

Comment:
This doesn't seem to go through all the nodes in order to search for content.
I added a text node randomly inside the tree and I couldn't find it.
Having the text in a canvas note instead appears to work fine.

Reply:
Do you have further information?

I tried the same and it does work in my case. Random Text note inside my tree:

![image](https://github.com/user-attachments/assets/8a3c281c-5d7e-4691-b002-917b260be4a7)

Reply:
@CobriMediaJulien , for the mind map you would need to add more nodes. From what I see in the screenshot it's only a single node ("now a mindmaptest"). Try pressing enter a few times to create a few subnodes and look at that subnode instead.

I believe you need to iterate recursively through all the nodes, similarly to how you do for canvas note.

Reply:
Could reproduce. I just search for topic value and not children. I will fix it this weekend.

Reply:
Fixed and commited to this PR

Reply:
@CobriMediaJulien , don't forget to remove the draft status from the PR once the required changes were done.

Reply:
this PR provides also a fix for:

https://github.com/TriliumNext/Notes/issues/772